src/doc/specifying-dependencies.md

```diff
@@ -0,0 +1,305 @@
+[dependencies]
+hello_utils = { path = "hello_utils", version = "0.1.0" }
+```
+
+# Overriding Dependencies
+
+Sometimes you may want to override one of Cargo’s dependencies. For example,
+let’s say you’re working on a project,
+[`uuid`](https://crates.io/crates/uuid), which depends on
+the package [`rand`](https://crates.io/crates/rand). You find a bug in `rand`, and
+you want to write a patch and be able to test out your patch by using your
+version of `rand` in `uuid`. Here’s what `uuid`’s `Cargo.toml` looks like:
+
+```toml
+[package]
+name = "uuid"
+version = "0.2.2"
+authors = ["The Rust Project Developers"]
+
+[dependencies]
+rand = { version = "0.3", optional = true }
+```
+
+You check out a local copy of `rand`, let’s say in your `~/src` directory:
+
+```shell
+$ cd ~/src
+$ git clone https://github.com/rust-lang-nursery/rand.git
+```
+
+You’d like to have `uuid` use your local version of `rand`,
+rather than the one on [crates.io], while you fix the bug.
+
+Cargo solves this problem by allowing you to have a local configuration
+that specifies an **override**. If Cargo finds this configuration when
+building your package, it will use the override on your local machine
+instead of the source specified in your `Cargo.toml`.
+
+Cargo looks for a directory named `.cargo` up the directory hierarchy of
+your project. If your project is in `/path/to/project/uuid`,
+it will search for a `.cargo` in:
+
+* `/path/to/project/uuid`
+* `/path/to/project`
+* `/path/to`
+* `/path`
+* `/`
+
+This allows you to specify your overrides in a parent directory that
+includes commonly used packages that you work on locally and share them
+with all projects.
+
+To specify overrides, create a `.cargo/config` file in some ancestor of
+your project’s directory (common places to put it is in the root of
+your code directory or in your home directory).
+
+Inside that file, put this:
+
+```toml
+paths = ["/path/to/project/rand"]
+```
+
+This array should be filled with directories that contain a `Cargo.toml`. In
+this instance, we’re just adding `rand`, so it will be the only one that’s
+overridden. This path must be an absolute path.
+
+Note: using a local configuration to override paths will only work for crates
+that have been published to [crates.io]. You cannot use this feature to tell Cargo
+how to find local unpublished crates.
+
+More information about local configuration can be found in the [configuration
+documentation](config.html).
+
+# Platform specific dependencies
+
+
+Platform-specific dependencies take the same format, but are listed under a
+`target` section. Normally Rust-like `#[cfg]` syntax will be used to define
+these sections:
+
+```toml
+[target.'cfg(windows)'.dependencies]
+winhttp = "0.4.0"
+
+[target.'cfg(unix)'.dependencies]
+openssl = "1.0.1"
+
+[target.'cfg(target_pointer_width = "32")'.dependencies]
+native = { path = "native/i686" }
+
+[target.'cfg(target_pointer_width = "64")'.dependencies]
+native = { path = "native/i686" }
```

Comment:
Yup, I moved it from [here](https://github.com/rust-lang/cargo/pull/2688/files#diff-0d7e52796da168d3e2f6b7a8100eb8dbL179), I'm happy to fix it though!